cg/apps/demultiplex/sample_sheet/validate.py

```diff
@@ -5,18 +5,18 @@
 from typing_extensions import Literal
 
 from cg.apps.demultiplex.sample_sheet.base import (
-    Sample,
+    NovaSeqSample,
     SampleSheet,
-    NovaSeqSampleBcl2Fastq,
-    NovaSeqSampleDragen,
+    SampleBcl2Fastq,
+    SampleDragen,
     SampleSheetError,
 )
 
 LOG = logging.getLogger(__name__)
 
 
-def validate_unique_sample(samples: List[Sample]) -> None:
-    """Validate that each sample only exists once"""
+def are_samples_unique(samples: List[NovaSeqSample]) -> bool:
```

Comment:
I think validation functions are intended to raise errors when a rule is broken, and return nothing if not. So the original structure is correct. But the function name could be clarified to `validate_samples_are_unique`

Reply:
Great this was one of my doubts, but it makes sense. My motivation was to make it more testable. But I will change it back to validation.